### PR TITLE
SW-5635 Add internal comments for non-numbered sections

### PR DESCRIPTION
**Pull request by nickgraz**
- Only applies in edit mode
- Hook up to 'update workflow details' flow

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2878?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2878** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2878?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2876** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2876?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/scenes/AcceleratorRouter/Deliverables/VariableInternalComment.tsx [hunk 1]
```diff
@@ -6,23 +6,31 @@ import TextField from '@terraware/web-components/components/Textfield/Textfield'
...
 
+  const variableValues = variable?.variableValues || [];
+
+  // For section variables, multiple variableValues are returned, so we need to find the one with the current ID
+  let variableValue: VariableValue | undefined;
+  if (variable.type === 'Section') {
+    variableValue = (variable?.variableValues || []).find((value) => value.variableId === variable.id);
+  } else {
+    variableValue = variableValues[0];
+  }
```
Comment (joshbuchea) [inline src/scenes/AcceleratorRouter/Deliverables/VariableInternalComment.tsx:29]: This is good to know, thanks for the comment 💯

**Review by joshbuchea — APPROVED** [review]

File: src/scenes/AcceleratorRouter/Deliverables/VariableInternalComment.tsx (hunk 1 above)
Comment (nickgraz) [reply src/scenes/AcceleratorRouter/Deliverables/VariableInternalComment.tsx:29]: The whole variables with values that are variable values that may also have values thing gets pretty heady pretty quick 😅

**Comment by nickgraz** [comment]
### Merge activity

* **Jul 18, 5:37 PM EDT**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2878).

**Pull request merged**